File: front-end/src/classes.ts
export class InvoiceItem {
  public invoice_id?: number;
  public invoice_number: number;
  public date: Date;
  public program: string;
  public amount: number;
  // public checked: boolean;
  public content: string;
  public image: string;
  public invoice_type?: string;

  constructor({
    date,
    program: title,
    amount,
    // checked,
    content,
    image,
    invoice_id,
    invoice_number,
  }: {
    date: Date;
    program: string;
    amount: number;
    // checked: boolean,
    content: string;
    image: string;
    invoice_id: number;
    invoice_number:number,
  }) {
    this.date = date;
    this.program = title;
    this.amount = amount;
    // this.checked = checked;
    this.content = content;
    this.image = image;
    this.invoice_id = invoice_id;
    this.invoice_number = invoice_number;
  }
}

File: classes.py
class invoice_content:
    date: str
    program: str
    amount: str
    content: str
    invoice_type: str
    invoice_number: int

    def __init__(
        self,
        date: str,
        program: str,
        amount: str,
        content: str,
        invoice_type: str,
        invoice_number: int,
    ):
        self.date = date
        self.program = program
        self.amount = amount
        self.content = content
        self.invoice_type = invoice_type
        self.invoice_number = invoice_number

    def to_dict(self):
        return {
            "date": self.date,
            "program": self.program,
            "amount": self.amount,
            "content": self.content,
            "invoice_type": self.invoice_type,
            "invoice_number": self.invoice_number,
        }


class api_invoice_return:
    success = True
    content: invoice_content

    def __init__(self, content: invoice_content):
        self.content = content

    def to_dict(self):
        return {
            "success": self.success,
            "content": self.content.to_dict(),
        }


class api_invoice_error:
    success = False
    error: str

    def __init__(self, error: str):
        self.error = error

    def to_dict(self):
        return {
            "success": self.success,
            "error": self.error,
        }
